lib/datadog/appsec/actions_handler/serializable_backtrace.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Datadog
+  module AppSec
+    module ActionsHandler
+      # This module serves encapsulates MessagePack serialization for caller locations.
+      #
+      # It serializes part of the stack:
+      # up to 32 frames (configurable)
+      # keeping frames from top and bottom of the stack (75% to 25%, configurable).
+      #
+      # It represents the stack trace that is added to span metastruct field.
+      class SerializableBacktrace
+        def initialize(locations:, stack_id:)
+          @stack_id = stack_id
+
+          max_depth = Datadog.configuration.appsec.stack_trace.max_depth
+          top_percent = Datadog.configuration.appsec.stack_trace.top_percentage
+
+          drop_from_idx = max_depth * top_percent / 100
+          drop_until_idx = locations.size - (max_depth - drop_from_idx)
+
+          frame_idx = -1
+          @serializable_locations_map = locations.each_with_object({}) do |location, map|
+            # we are dropping frames from library code without increasing frame index
+            next if location.path&.include?('lib/datadog')
+
+            frame_idx += 1
+
+            next if frame_idx >= drop_from_idx && frame_idx < drop_until_idx
+
+            map[frame_idx] = location
+          end
```

Comment:
I think this should be done only when needed and we can keep initializer clean and only about settings.

Reply:
fixed [0d8e709](https://github.com/DataDog/dd-trace-rb/pull/4526/commits/0d8e70904322dba2fc64922f350f9c41731f0a53)